device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt

```diff
@@ -185,21 +192,29 @@ public class CalendarService : PluginRegistry.RequestPermissionsResultListener {
         return null;
     }
 
-    public fun retrieveEvents(calendarId: String) {
+    public fun retrieveEvents(calendarId: String, startDate: Long, endDate: Long) {
         _calendarId = calendarId;
+        _calendarEventsStartDate = startDate;
+        _calendarEventsEndDate = endDate;
         if (ensurePermissionsGranted(REQUEST_CODE_RETRIEVE_EVENTS)) {
             val calendar = retrieveCalendar(calendarId, true);
             if (calendar == null) {
                 finishWithError(CALENDAR_RETRIEVAL_FAILURE, "Couldn't retrieve the Calendar with ID ${calendarId}");
                 return;
             }
+            if (startDate > endDate) {
+                finishWithError(EVENTS_RETRIEVAL_FAILURE, EVENTS_START_DATE_LARGER_THAN_END_DATE_MESSAGE);
```

Comment:
hmm good point, though i think maybe we should catch this at the Dart level and throw an exception there as it'll likely be more performant that way. can be done later though, just a thought